tests/settings_tests/tests.py

```diff
@@ -511,3 +511,45 @@ def test_session_verification_deprecation_neither(self):
             warnings.filterwarnings('always')
             Settings('fake_settings_module')
         self.assertEqual(len(warn), 0)
+
+
+class TestAllowedHostsSettings(unittest.TestCase):
+    """
+    Make sure Allowed_Hosts setting raises an ImproperlyConfigured during
+    appropriate conditions.
+    """
+
+    def test_debug_off_unconfigured(self):
+        settings_module = ModuleType('fake_settings_module')
+        settings_module.DEBUG = False
+        settings_module.ALLOWED_HOSTS = []
+        with self.assertRaises(ImproperlyConfigured):
```

Comment:
it's helpful to use assertRaisesMessage to ensure this is the exception we expect